SportsGoods.Core/Interfaces/IProductRepository.cs

```diff
@@ -0,0 +1,16 @@
+﻿using SportsGoods.Core.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SportsGoods.Core.Interfaces
+{
+    public interface IProductRepository
+    {
+        IQueryable<Product> Products { get; }
```

Comment:
No IQueryable in Repositories 

In this post I am going to make the case why returning IQueryable from a Repository Pattern is an anti-pattern and why you should not do it.

https://weblogs.asp.net/dotnetstories/repository-iqueryable

Reply:
Done. Refactored IProductRepository and ProductRepository. The interface no longer implements IQueryable.